src/components/tabBars.js

```diff
@@ -123,17 +124,23 @@ TabBar.propTypes = {
  * @param tabStyle Optionally, additional styles to add to each tab
  * @param children Children, which will be appended to teh end of the tab bar
  * @param props Any additional properties to add to the container menu element
+ * @param metricsPrefix Optionally, a metrics event will be fired with this prefix and the name of the tab whenever the given tab is viewed.
+ *    The form of the event name will be <metricsPrefix>:view:<tabName> (any spaces in tabName will be replaced with "-").
+ * @param metricsData Optionally, if metricsPrefix is supplied then this data object will be included with the event that is fired.
  */
 export const SimpleTabBar = ({
-  value, onChange, tabs, tabProps = {}, panelProps = {}, style = {}, tabStyle = {}, children, ...props
+  value, onChange, tabs, tabProps = {}, panelProps = {}, style = {}, tabStyle = {}, metricsPrefix = undefined, metricsData = {}, children, ...props
 }) => {
   useLabelAssert('SimpleTabBar', props)
-
   const tabIds = _.map(useUniqueId, _.range(0, tabs.length))
   const panelRef = useRef()
+  const emitViewMetric = key => {
+    !!metricsPrefix && Ajax().Metrics.captureEvent(`${metricsPrefix}:view:${_.replace(/\s/g, '-', key)}`, metricsData)
+  }
```

Comment:
Since the name of this isn't `maybeEmitViewMetric`, I might:
```suggestion
  const emitViewMetric = key => {
    Ajax().Metrics.captureEvent(`${metricsPrefix}:view:${_.replace(/\s/g, '-', key)}`, metricsData)
  }
```

Reply:
If I do that, I have to be careful not to call `emitViewMetric` if the prefix key has not been provided (which is what I'm using to "opt-in" to the metrics). I will change the name to `maybeEmitViewMetric`.